core/src/main/java/org/apache/iceberg/GenericDataFile.java

```diff
@@ -26,7 +26,7 @@
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
 import org.apache.iceberg.types.Types;
 
-class GenericDataFile extends BaseFile<DataFile> implements DataFile {
+public class GenericDataFile extends BaseFile<DataFile> implements DataFile {
```

Comment:
This and GenericDeleteFile shouldn't be made public, based on discussion with @rahil-c this is just an artifact of having the parser implementation in the REST package which can't access this class. I think the real solution should be to uplevel the new parser to core + reuse as much of the existing Content file parsers

Reply:
If we do my suggestion below, this along with the rev API changes should go away and get the diff to be a lot smaller by reusing ContentFileParser as it is

Reply:
Will fix this thanks @amogh-jahagirdar